infrastructure/tooling/src/changelog/index.js

```diff
@@ -129,10 +153,134 @@ class ChangeLog {
   }
 }
 
-const main = async (options) => {
+const check_pr = async (pr) => {
+  const files = [];
+  let url = `https://api.github.com/repos/taskcluster/taskcluster/pulls/${pr}/files`;
+  while (1) {
+    const res = await request
+      .get(url)
+      .set('user-agent', 'taskcluster-ci');
+    res.body.forEach(({filename}) => files.push(filename));
+    const link = res.get('link');
+    if (link) {
+      const parsed = parseLinkHeader(link);
+      if (parsed.next) {
+        url = parsed.next.url;
+        console.log(url);
+        continue;
+      }
+    }
+
+    break;
+  }
+
+  // files that do not require a changelog entry if they are the only thing changed.  This
+  // is similar to the list in .gitattributes., along with yarn and package.json
+  const boringFiles = [
+    /yarn\.lock$/,
+    /package\.json$/,
+    /^\.yarn$/,
+    /^\.taskcluster.yml/,
+    /^infrastructure\//,
+    /^generated\//,
+    /^clients\/client-web\/src\/clients\//,
+    /^clients\/client-py\/taskcluster\/generated\//,
+    /^clients\/client-py\/README\.md$/,
+    /^clients\/client\/src\/apis\.js$/,
+    /^services\/*\/Procfile$/,
+    /^clients\/client-shell\/apis\/services\.go$/,
+    /^dev-docs\/dev-config-example\.yml$/,
+    /^clients\/client-go\/tc*\//,
+  ];
+  const hasImportantFiles = files.some(filename => boringFiles.every(r => !r.test(filename)));
+
+  if (!hasImportantFiles) {
+    console.log(`${chalk.bold.green(`PR ${pr} OK:`)} does not contain any changes requiring a changelog`);
+    return true;
+  }
+
+  const hasChangelog = files.some(filename => filename.startsWith('changelog/'));
+
+  if (hasImportantFiles && !hasChangelog) {
+    console.log(`${chalk.bold.red('ERROR:')} Pull Request ${pr} does not modify any files in 'changelog/'`);
+    return false;
+  }
+  console.log(chalk.bold.green(`PR ${pr} OK`));
+  return true;
+};
+
+const add = async (options) => {
+  let level;
+  if (options.major) {
+    level = 'major';
+  } else if (options.minor) {
+    level = 'minor';
+  } else if (options.patch) {
+    level = 'patch';
+  } else if (options.silent) {
+    level = 'silent';
+  } else {
+    console.log('Must specify one of --major, --minor, --patch, or --silent');
+    process.exit(1);
+  }
+
+  let name, reference;
+  if (options.issue) {
+    name = `issue-${options.issue}`;
+    reference = `reference: issue ${options.issue}\n`;
+  } else if (options.bug) {
+    name = `bug-${options.bug}`;
+    reference = `reference: bug ${options.bug}\n`;
+  } else {
+    name = taskcluster.slugid();
```

Comment:
Maybe we make this an option you have to explicitly ask for. I'd like to be able to run this and have it say, "you should either reference a bug, issue, or ask for a generated id".